app/controllers/candidate_interface/candidate_interface_controller.rb

```diff
@@ -4,6 +4,7 @@ class CandidateInterfaceController < ApplicationController
     before_action :authenticate_candidate!
     before_action :set_user_context
     before_action :check_cookie_preferences
+    before_action :check_account_locked
```

Comment:
This works well. I wonder if we should not even send the sign-in email if the account is locked?

Reply:
@duncanjbrown We could check from the sign-up/in page whether the given email was for a locked account and redirect straight to the account blocked page. My only issue with this is that it gives unauthenticated users a way to find out whether a given email address is registered and locked. If we want to ensure that a candidate has to be authenticated to find out whether they have been locked out or not then we probably shouldn't do this.

Reply:
yes i think it's fine